modules/ROOT/pages/custom-policy-set-authentication-example.adoc

```diff
@@ -0,0 +1,198 @@
+= Developing a Custom Policy Using the Event Authentication Extension
+ifndef::env-site,env-github[]
+include::_attributes.adoc[]
+endif::[]
+
+You can plan, develop, and implement a custom policy based on the Set Authentication Policy example described in this topic.
+
+The example implements the Event Authentication Extension to propagate a Client ID property,  using the authentication data of the Mule Event. Creating a policy involves the following steps:
+
+. Evaluate your use case based on the <<use_case, use case>> provided.
+. <<set-up-the-project, Set up your project>>.
+. <<create_the_YAML, Create the YAML Configuration File>>.
+. <<implement_the_policy, Implement your policy>>.
+. <<upload_to_exchange, Upload your policy to Exchange>>.
+. <<apply_custom_policy, Apply your custom policy>>.
+
+[[use_case]]
+== The Use Case
+
+ProTeams is a sports management application that enables its users to manage team and sports activities. ProTeams users can perform actions on behalf of their teams, such as participating in chats with other teams, organizing exhibitions or friendly matches,and confirming assistance on tournament dates.
+
+Each team has a client application in Anypoint Platform, and uses a Client ID Enforcement policy to validate requests to their site. To keep the team management tasks hidden from the players, separate IDs are created for them. 
+
+The team owners can proxy tasks for the players, if required, using the unique player IDs and team client ID mapping. For example, if the team management wants to perform analytics on each player for the previous game or set up matches for selective players,they can do so using the mapping.
+
+To create this player ID and team client ID mapping, the team owners need to develop a new custom policy that retrieves the user IDs from a header called `userId`on the request. The user IDs can also be obtained using query parameters.
+
+The policy then performs the mapping, and injects the obtained client ID into the authentication data of the Mule event. This event is then retrieved by the Client ID Enforcement policy for performing analytics, and so on.
+
+ProTeams uses the Set Authentication custom policy example as a template and then add their own mapping logic to it.
```

Comment:
```suggestion
ProTeams uses the Set Authentication custom policy example as a template and then adds their own mapping logic to it.
```

Reply:
I had a bit of difficulty following the scenario. I wonder if we can somehow illustrate it in future releases?

Reply:
Sure! Thanks!